cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -1580,266 +1433,213 @@ impl Node<Option<cst::Member>> {
         }
     }
 
-    fn to_ref_or_refs<T: RefKind>(&self, errs: &mut ToASTErrors, var: ast::Var) -> Option<T> {
-        // if `self` doesn't have data, nothing we can do here, just propagate
-        // the `None`; we don't need to signal an error, because one was already
-        // signaled when the `Node` without data was created
-        let mem = self.as_inner()?;
+    fn to_ref_or_refs<T: RefKind>(&self, var: ast::Var) -> Result<T> {
+        let mem = self.try_as_inner()?;
 
         match mem.access.len() {
-            0 => mem.item.to_ref_or_refs::<T>(errs, var),
+            0 => mem.item.to_ref_or_refs::<T>(var),
             _n => {
-                errs.push(self.to_ast_err(ToASTErrorKind::wrong_node(T::err_str(), "a `.` expression", Some("entity types and namespaces cannot use `.` characters -- perhaps try `_` or `::` instead?"))));
-                None
+                Err(self.to_ast_err(ToASTErrorKind::wrong_node(T::err_str(), "a `.` expression", Some("entity types and namespaces cannot use `.` characters -- perhaps try `_` or `::` instead?"))).into())
             }
         }
     }
 
-    fn to_expr_or_special(&self, errs: &mut ToASTErrors) -> Option<ExprOrSpecial<'_>> {
-        // if `self` doesn't have data, nothing we can do here, just propagate
-        // the `None`; we don't need to signal an error, because one was already
-        // signaled when the `Node` without data was created
-        let mem = self.as_inner()?;
+    fn to_expr_or_special(&self) -> Result<ExprOrSpecial<'_>> {
+        let mem = self.try_as_inner()?;
 
-        let maybe_prim = mem.item.to_expr_or_special(errs);
+        let maybe_prim = mem.item.to_expr_or_special();
+        let maybe_accessors = ParseErrors::transpose(mem.access.iter().map(|a| a.to_access()));
 
-        // collect() allows all conversions to run and generate errors
-        let mut accessors: Vec<_> = mem.access.iter().map(|a| a.to_access(errs)).collect();
+        // Return errors in case parsing failed for any element
+        let (prim, mut accessors) = flatten_tuple_2(maybe_prim, maybe_accessors)?;
 
-        // we use `head` as our failure indicator going forward
-        let mut head = maybe_prim;
-        // we need at least three available items, for example:
-        // var .call (args) -  which becomes one expr
-        // so we use slice matching
+        // `head` will store the current translated expression
+        let mut head = prim;
+        // `tail` will store what remains to be translated
         let mut tail = &mut accessors[..];
 
-        // Starting off with a failure and filtering items from the accessor list
-        // can cause false error messages. We consider this acceptable for now because
-        // they only occur along side a real error.
-        // TODO(#439): eliminate the false errors (likely with `Option`s inside `AstAccessor`)
```

Comment:
Since you `break` immediately on an error now I feel like this issue might actually be fixed. 

Regardless, I don't have a good idea of what the issue was to begin with, so I'm in favor of closing.